tests/Doctrine/Tests/DBAL/Platforms/DB2PlatformTest.php

```diff
@@ -377,6 +377,33 @@ public function testModifiesLimitQuery()
         );
     }
 
+    public function testModifiesLimitQueryWithOrderBy()
+    {
+        self::assertEquals(
+            'SELECT * FROM user ORDER BY id',
+            $this->_platform->modifyLimitQuery('SELECT * FROM user ORDER BY id', null, null)
+        );
+
+        self::assertEquals(
+            'SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER(ORDER BY id) AS DC_ROWNUM FROM (SELECT * FROM user) db21) db22 WHERE db22.DC_ROWNUM <= 10',
+            $this->_platform->modifyLimitQuery('SELECT * FROM user ORDER BY id', 10, 0)
+        );
+
+        self::assertEquals(
+            'SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER(ORDER BY id) AS DC_ROWNUM FROM (SELECT * FROM user) db21) db22 WHERE db22.DC_ROWNUM <= 10',
+            $this->_platform->modifyLimitQuery('SELECT * FROM user', 10)
```

Comment:
Doesn't this case cover the same execution path as the previous one? The implementation checks `if ($offset > 0)`, so there's no difference between `0` and `NULL`. And from the SQL standpoint, no offset means "start from zero".

Reply:
I copied form the test case above and added the order by

Reply:
I think this test shouldn't focus on testing of different variations of offset and limit since its logic doesn't depend on those, but instead, focus on edge cases of the `ORDER BY` expression.

Reply:
Good point... Will see what I can come with.
Everyone else is more than welcome to chime in 😄

Reply:
Since the regexp matches the last `ORDER BY` expression in the string (as far as I understand, this is to ignore sub-queries), there should be a test covering a query with a sub-query.

Reply:
The test will be there as soon as I cover more edge cases.

Reply:
Did I mention I don't like "edge cases" 🤣 
Can someone introduce more parts to this class, so we can access certain items before the SQL string is build?

Reply:
Before we proceed with the parser, could you explain what exactly a non-empty `OVER()` is needed in this case? As far as I understand, if it's empty, then the row number assigned in the order specified by `ORDER BY` which is exactly what's needed for implementing `LIMIT, OFFSET`. Doesn't it work in all cases, or it works better with a non-empty `OVER()`.

Reply:
On IBMDB2 everything is a little bit more complicated (very unfortunate). The row_number is assigned based on what's in OVER(). If OVER() is empty, the row_number is based on the relative-record-number of the table. The result of your regular order by would not look like what you are expecting if used with row_number.
If you need, as in most cases, the row_number assignment based on your order_by, you need the order in OVER() as well.
Either way, I agree, the current test-cases don't make much sense.

Reply:
@guidofaecke thanks for the clarification. With the above said and the fact that the row number in this query is used for emulating `LIMIT`, let's start with a test case in which a selected subset of the rows doesn't match the expectation.